homeassistant/components/sensor/nmbs.py

```diff
@@ -0,0 +1,217 @@
+    station_live = config[CONF_STATION_LIVE]
+
+    sensors = [NMBSSensor(name, station_from, station_to, api_client)]
+
+    if station_live is not None:
+        sensors.append(NMBSLiveBoard(station_live, api_client))
+
+    add_entities(sensors, True)
+
+
+class NMBSLiveBoard(Entity):
+    """Get the next train from a station's liveboard."""
+
+    def __init__(self, live_station, api_client):
+        """Initialize the sensor for getting liveboard data."""
+        self._station = live_station
+        self._api_client = api_client
+        self._attrs = {}
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the sensor default name."""
+        return DEFAULT_NAME_LIVE
+
+    @property
+    def icon(self):
+        """Return the default icon or an alert icon if delays."""
+        if self._attrs is not None and int(self._attrs['delay']) > 0:
+            return DEFAULT_ICON_ALERT
+
+        return DEFAULT_ICON
+
+    @property
+    def state(self):
+        """Return sensor state."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the sensor attributes if data is available."""
+        if self._state is None or self._attrs is None:
+            return None
+
+        delay = get_delay_in_minutes(self._attrs["delay"])
+        departure = get_time_until(self._attrs['time'])
+
+        attrs = {
+            'departure': "In {} minutes".format(departure),
+            'extra_train': int(self._attrs['isExtra']) > 0,
+            'occupancy': self._attrs['occupancy']['name'],
+            'vehicle_id': self._attrs['vehicle'],
+            ATTR_ATTRIBUTION: "https://api.irail.be/",
+        }
+
+        if delay > 0:
+            attrs['delay'] = "{} minutes".format(delay)
+
+        return attrs
+
+    def update(self):
+        """Set the state equal to the next departure."""
+        liveboard = self._api_client.get_liveboard(self._station)
+        next_departure = liveboard['departures']['departure'][0]
+
+        self._attrs = next_departure
+        self._state = "Track {} - {}".format(
+            next_departure['platform'], next_departure['station'])
+
+
+class NMBSSensor(Entity):
+    """Get the the total travel time for a given connection."""
+
+    def __init__(self, name, station_from, station_to, api_client):
+        self._name = name
+        self._station_from = station_from
+        self._station_to = station_to
+        self._api_client = api_client
+        self._attrs = {}
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return 'min'
+
+    @property
+    def icon(self):
+        """Return the sensor default icon or an alert icon if any delay."""
+        delay = get_delay_in_minutes(self._attrs['departure']['delay'])
```

Comment:
Can `self._attrs` be `None` at this point? We're checking for it below. If it could we would get an error here.